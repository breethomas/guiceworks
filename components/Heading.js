// @flow
import React from 'react'
import css, { media2 } from '../styles/css'
import SerifBold from './SerifBold'

type Props = {
  tag?: string,
}

const style = css(
  { fontSize: 40 },
  media2({ fontSize: 74 }),
)

const Heading = ({ tag, ...props }: Props) => (
  <SerifBold className={style} tag={tag} {...props} />
)

Heading.defaultProps = {
  tag: 'h2',
}

export default Heading
