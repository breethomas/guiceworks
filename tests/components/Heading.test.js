import React from 'react'
import renderer from 'react-test-renderer'
import Heading from '../../components/Heading'

describe('Heading', () => {
  it('shows the correct snapshot for the Heading component tree (default)', () => {
    const component = <Heading>Heading</Heading>
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('shows the correct snapshot for the Heading component tree (tag)', () => {
    const component = <Heading tag="span">Heading</Heading>
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
