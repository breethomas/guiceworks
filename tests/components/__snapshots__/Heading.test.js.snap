// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading shows the correct snapshot for the Heading component tree (default) 1`] = `
<h2
  className="css-lcomo css-1g95qkz"
>
  Heading
</h2>
`;

exports[`Heading shows the correct snapshot for the Heading component tree (tag) 1`] = `
<span
  className="css-lcomo css-1g95qkz"
>
  Heading
</span>
`;
